### PR TITLE
Updated the dependencies.

### PR DESCRIPTION
**Pull request by kittaakos**
- Build all three platforms on Travis.
 - Adjusted VS Code settings.

Signed-off-by: Akos Kitta <kittaakos@typefox.io>

### CODE REVIEW

**Comment by AlexTugarev** [comment]
one test is failing on win :-(

**Comment by kittaakos** [comment]
> one test is failing on win :-(

wow, the `master` state has failed on my Windows 10 image too.

**Comment by AlexTugarev** [comment]
> wow, the master state has failed on my Windows 10 image too.

without any changes?

**Comment by kittaakos** [comment]
> without any changes?

Yes, indeed weird. On my local Windows image, both the `master` and this PR have failed. But the AppVeyor CI is green on the `master`.

I started to investigate [here](https://github.com/theia-ide/dugite-extra/pull/32) and locally in parallel. I aligned the Node.js, Git, and TS versions; the CI passes, the local test fails.

File: src/command/stage.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export async function unstage(
         args.push(...['rm', '--cached', '-r', '--']);
     } else {
         if (_where === 'working-tree') {
-            args.push(...['checkout-index', '-f']);
+            args.push(...['checkout-index', '-f', '-u']);
```
Comment (kittaakos) [inline src/command/stage.ts]: Otherwise, it fails with a `git checkout-index: HEAD is not in the cache` error.

Comment (kittaakos) [reply src/command/stage.ts]: Also: https://stackoverflow.com/questions/43239029/what-does-git-checkout-index-u-or-index-do

**Comment by kittaakos** [comment]
The AppVeyor builds are red as we switched to Travis. I am merging it.

**Pull request merged**